lib/Runtime/Base/ScriptContext.cpp

```diff
@@ -837,6 +837,18 @@ namespace Js
         return propertyRecord->GetPropertyId();
     }
 
+    void ScriptContext::GetOrAddPropertyRecord(Js::JavascriptString * propertyString, PropertyRecord const** propertyRecord)
+    {
+        if (VirtualTableInfo<Js::PropertyString>::HasVirtualTable(propertyString) && propertyString->GetScriptContext() == this)
```

Comment:
at certain places we have checks for         ```VirtualTableInfo<Js::LiteralStringWithPropertyStringPtr>``` do you need to handle that part as well?

Reply:
https://github.com/Microsoft/ChakraCore/pull/3875